cloudimg/ms_azure.py

```diff
@@ -285,11 +287,15 @@ def filter_object_by_tags(self, tags):
                 blobs_list = container_client.find_blobs_by_tags(
                     filter_expression=filter_expression
                 )
-                try:
-                    return blobs_list.next()
-                except StopIteration:
-                    continue
-        return
+                for tag in tags.keys():
+                    log.debug("Filtering the blob with the tag: %s" % tag)
```

Comment:
How exactly do you know that we are currently filtering on `tag` ?  Do you know that the order of `blobs_list` corresponds to the order of `tags.keys()`?  How do you know that?